subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -1001,36 +1001,119 @@ void ull_cp_tx_ack(struct ll_conn *conn, struct node_tx *tx)
 
 void ull_cp_rx(struct ll_conn *conn, struct node_rx_pdu *rx)
 {
+	struct proc_ctx *ctx_l;
+	struct proc_ctx *ctx_r;
 	struct pdu_data *pdu;
-	struct proc_ctx *ctx;
+	bool unexpected_l;
+	bool unexpected_r;
 
 	pdu = (struct pdu_data *)rx->pdu;
 
-	if (!pdu_is_terminate(pdu)) {
-		/*
-		 * Process non LL_TERMINATE_IND PDU's as responses to active
-		 * procedures
-		 */
+	if (pdu_is_terminate(pdu)) {
```

Comment:
ok, so I think invalid PDU check could be hooked before this check and would also mark both contexts as NULL and in addition llcp_rr_new() would be given bool parameter to indicate PDU was invalid and unknown PDU proc needs to be started